elastic/apm-server
# aggregation/txmetrics: include transaction outcome

## Motivation/summary

Add `Transaction.Outcome` to aggregation key, and add `event.outcome` to the metrics docs produced.

## Checklist


I have considered changes for:
~- [ ] documentation~
~- [ ] logging (add log lines, choose appropriate log selector, etc.)~
~- [ ] metrics and monitoring (create issue for Kibana team to add metrics to visualizations, e.g. [Kibana#44001](https://github.com/elastic/kibana/issues/44001))~
~- [ ] telemetry~
~- [ ] Elasticsearch Service (https://cloud.elastic.co)~
~- [ ] Elastic Cloud Enterprise (https://www.elastic.co/products/ece)~
~- [ ] Elastic Cloud on Kubernetes (https://www.elastic.co/elastic-cloud-kubernetes)~

## How to test these changes

1. Start apm-server with transaction metrics aggregation enabled
2. Send some transactions with differing outcomes (e.g. using apm-agent-python)
3. Observe multiple metrics are produced with the same outcome values

## Related issues

Closes #4041
